pqdict/__init__.py

```diff
@@ -30,64 +29,87 @@
 
 Documentation at <http://pqdict.readthedocs.org/en/latest>.
 
-:copyright: (c) 2012-2015 by Nezar Abdennur.
+:copyright: (c) 2012-2023 by Nezar Abdennur.
 :license: MIT, see LICENSE for more details.
 
 """
-try:
-    from collections.abc import MutableMapping as _MutableMapping
-except ImportError:
-    # 2.7 compatability
-    from collections import MutableMapping as _MutableMapping  # type: ignore
-
-from six.moves import range
-from operator import lt, gt
-
+from collections.abc import Iterable, Mapping, MutableMapping
+from operator import gt, lt
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    Iterable,
+    Iterator,
+    List,
+    Mapping,
+    Optional,
+    Self,
```

Comment:
`typing.Self` was only added in Python 3.11. You can use `typing_extensions.Self` or you can do a custom type https://peps.python.org/pep-0673/